### PR TITLE
Remove Starkware support

### PR DESCRIPTION
**Pull request by apaillier-ledger**
## Description

Remove old Starkware internal "plugin" from the app.
Saves about 8 KB of flash.

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Tests
- [ ] Documentation
- [x] Other (for changes that might not fit in any category)

### CODE REVIEW

File: src_plugins/erc20/erc20_plugin.c [hunk 1]
```diff
@@ -74,19 +74,6 @@ bool check_contract(erc20_parameters_t *context) {
...
-            case STARK_QUANTUM_ERC20:
-            case STARK_QUANTUM_MINTABLE_ERC20:
-                return true;
-            default:
-                return false;
-        }
-    }
-#endif
     return true;
 }
```
Comment (tdejoigny-ledger) [inline src_plugins/erc20/erc20_plugin.c:78]: is this function useful (now) ?

File: tests/speculos/test_configuration_cmd.py [hunk 2]
```diff
@@ -1,10 +1,3 @@
...
 def test_configuration(cmd):
-    if cmd.model == "nanos":
-        assert cmd.get_configuration() == (14, 1, 10, 4)
-
-    if cmd.model == "nanox":
-        assert cmd.get_configuration() == (14, 1, 10, 4)
-
-    if cmd.model == "nanosp":
-        assert cmd.get_configuration() == (14, 1, 10, 4)
+    assert cmd.get_configuration() == (2, 1, 10, 4)
```
Comment (tdejoigny-ledger) [inline tests/speculos/test_configuration_cmd.py:3]: will have to be updated or removed at some point (avoid hardcoded values here)

File: src_plugins/erc20/erc20_plugin.c (hunk 1 above)
Comment (apaillier-ledger) [reply src_plugins/erc20/erc20_plugin.c]: It has been removed :+1:

File: tests/speculos/test_configuration_cmd.py (hunk 2 above)
Comment (apaillier-ledger) [reply tests/speculos/test_configuration_cmd.py:3]: These tests will be ultimately replaced by Ragger tests :+1:

File: src/eth_plugin_internal.c [hunk 3]
```diff
@@ -27,83 +22,15 @@ const uint8_t* const ETH2_SELECTORS[NUM_ETH2_SELECTORS] = {ETH2_DEPOSIT_SELECTOR
...
-
 // All internal alias names start with 'minus'
 
 const internalEthPlugin_t INTERNAL_ETH_PLUGINS[] = {
-    {erc20_plugin_available_check,
-     (const uint8_t**) ERC20_SELECTORS,
-     NUM_ERC20_SELECTORS,
-     "-erc20",
-     erc20_plugin_call},
+    {NULL, (const uint8_t**) ERC20_SELECTORS, NUM_ERC20_SELECTORS, "-erc20", erc20_plugin_call},
```
Comment (tdejoigny-ledger) [inline src/eth_plugin_internal.c:28]: this structure should be cleaned ultimately

**Review by tdejoigny-ledger — APPROVED** [review]

**Pull request merged**